File: 30-Day LeetCoding Challenge/C++/canJump.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if(nums.size() == 1) return true;
        int maxJump = 1;
        for(int i = 0; i < nums.size(); i++) {
            maxJump--;
            if (maxJump == 0 && nums[i] == 0 && i != nums.size() - 1) {
                return false;
            }
            maxJump = max(maxJump, nums[i]);
        }
        return true;
    }
};

File: 30-Day LeetCoding Challenge/Python/canJump.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        length = len(nums)
        if length == 1:
            return True
        maxJump = 1
        for i in range(length):
            maxJump -= 1
            if maxJump == 0 and nums[i] == 0 and i != length - 1:
                return False
            maxJump = max(maxJump, nums[i])
        return True